packages/story-editor/src/components/checklist/checks/pageBackgroundLowTextContrast/check.js

```diff
@@ -344,6 +344,42 @@ function getBackgroundColorByType(element) {
       return () => undefined;
   }
 }
+/**
+ * Returns all failing pages as well as the offending elements' id.
+ *
+ * @param {Array<Page>}  storyPages The story's page to check for contrast failures
+ * @param {Object} pageSize pageWidth and pageHeight from useLayout state
+ * @param {Page} currentPage If we only care about checking a single currentPage
+ * @return {Array<Object>} Each object contains the page that failed and the results
+ * from `pageBackgroundTextLowContrast`. In `results` you will find the offending
+ * elements' id(s).
+ */
+export async function getFailingPages(storyPages, pageSize, currentPage) {
+  const promises = [];
+  (storyPages || []).forEach((page) => {
+    // If we only care about checking currentPage bail early
+    // In this case, getFailingPages will return a max of 1 page.
+    if (currentPage && currentPage.id !== page.id) {
+      return;
+    }
```

Comment:
💭 I think rather than updating to add a new argument, we could pass in the current page in its own array using the same function as before:

```js
const failedPages = getFailingPages([currentPage], pageSize);
const didCurrentPageFail = failedPages[0];
```

This reduces the surface area we'd need to test and keeps the function a little more simple. Thoughts?

Reply:
That makes total sense, I don't know why I made it harder than it should be.

Reply:
This was so obvious I seriously don't know why I did the other thing 🤦🏻‍♀️ Thanks so much @samwhale